src/ol_openedx_chat_xblock/block.py

```diff
@@ -0,0 +1,145 @@
+    """
+
+    display_name = String(
+        display_name=_("Display Name"),
+        default=_("OL Chat XBlock"),
+        scope=Scope.settings,
+        help=_(
+            "This name appears in the horizontal navigation at the top of the page."
+        ),
+    )
+    course_id = String(
+        default=0,
+        scope=Scope.settings,
+        help=_("Course ID of the relevant course in Canvas"),
+    )
+    editable_fields = ("display_name", "course_id")
+
+    def student_view(self, context=None):  # noqa: ARG002
+        """
+        Render the student view of the block.
+        """
+
+        html = "<div>THIS IS THE STUDENT VIEW OF THE CHAT XBLOCK</div>"
+        return Fragment(html)
+
+    @XBlock.handler
+    def ol_chat(self, request, suffix=""):  # noqa: ARG002
+        """Start the chat session via external AI API."""
+
+        try:
+            request_data = request.json
+        except Exception:  # noqa: BLE001
+            log.warning("Invalid JSON in chat request.")
+            return Response(
+                {"error": "Invalid request body. Expected JSON."},
+                status=api_status.HTTP_400_BAD_REQUEST,
+                content_type="application/json",
+            )
+
+        api_url = settings.MIT_LEARN_AI_XBLOCK_CHAT_API_URL
+        api_token = settings.MIT_LEARN_AI_XBLOCK_CHAT_API_TOKEN
+
+        if not api_url or not api_token:
+            log.error("Missing AI API configuration (URL or token).")
+            return Response(
+                {"error": "Missing API configurations. Please check your settings."},
+                status=api_status.HTTP_400_BAD_REQUEST,
+                content_type="application/json",
+            )
+
+        if not self.course_id:
+            log.error("Course ID is not set for the XBlock.")
+            return Response(
+                {"error": "Course ID is required."},
+                status=api_status.HTTP_400_BAD_REQUEST,
+                content_type="application/json",
+            )
+
+        message = request_data.get("message", "").strip()
+        if not message:
+            return Response(
+                {"error": "Message field is required."},
+                status=api_status.HTTP_400_BAD_REQUEST,
+                content_type="application/json",
+            )
+
+        payload = {
+            "collection_name": "content_files",
+            "message": message,
+            "course_id": self.course_id,
+        }
+
+        headers = {
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {api_token}",
+        }
+
+        try:
+            response = requests.post(api_url, json=payload, headers=headers, timeout=30)
+            response.raise_for_status()
+            return Response(
+                response.text,
+                status=response.status_code,
+                content_type=response.headers.get("Content-Type", "application/json"),
+            )
```

Comment:
@ChristopherChudzicki What type of response do we expect in the aiChat component when the request routes through the xBlock backend?